### PR TITLE
tests: build_all: ieee802154

### PR DESCRIPTION
**Pull request by JordanYates**
Add a build_all application for ieee802154 drivers.
Added as I noticed that CC1200 and CC2520 drivers were not being compiled in any in-tree code.
Fixes an old issue #11519.

### CODE REVIEW

File: tests/drivers/build_all/ieee802154/gpio.dtsi [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2020, Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Application overlay for gpio devices
+ */
+
+test_gpio_sm351lt: sm351lt0 {
+	compatible = "honeywell,sm351lt";
```
Comment (galak) [inline tests/drivers/build_all/ieee802154/gpio.dtsi:10]: why is this node needed for ieee802154?

File: tests/drivers/build_all/ieee802154/prj.conf [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
+CONFIG_GPIO=y
+CONFIG_SPI=y
+
+CONFIG_NETWORKING=y
+CONFIG_NET_L2_IEEE802154=y
+CONFIG_NET_L2_IEEE802154_SECURITY=y
+
+CONFIG_IEEE802154_CC1200=y
+CONFIG_IEEE802154_CC2520=y
```
Comment (galak) [inline tests/drivers/build_all/ieee802154/prj.conf:13]: Please enable build test for all 802.15.4 drivers (CC13xx, DW1000, KW41z, etc.)

File: tests/drivers/build_all/ieee802154/gpio.dtsi (hunk 1 above)
Comment (JordanYates) [reply tests/drivers/build_all/ieee802154/gpio.dtsi]: its not

File: tests/drivers/build_all/ieee802154/prj.conf (hunk 2 above)
Comment (JordanYates) [reply tests/drivers/build_all/ieee802154/prj.conf]: Done, requires adding many test configs though, as half of the drivers are for SoC peripherals, not external modems.

**Review by cfriedt — APPROVED** [review]

**Comment by cfriedt** [comment]
@nashif - can you please ack? This is tests-only and also fixes an old enhancement issue

**Review by gmarull — APPROVED** [review]

**Pull request merged**